.gitlab-ci.yml

```diff
@@ -0,0 +1,93 @@
+# SPDX-License-Identifier: AGPL-3.0+
+#
+# Copyright (C) 2018 Ultimaker B.V.
+# Copyright (C) 2018 Olliver Schinagl <oliver@schinagl.nl>
+#
+# GitLab CI package build is split up into two parts.
+# * Build environment: creates the docker image that is needed for package generation
+# * Package generation: creates the final artifact
+
+stages:
+- lint
+- prepare
+- verify
+- push
+- build
+- cleanup
+
+# Common requisites
+# ==========
+.parameters_template: &parameters_common
+  tags:
+  - embedded
+  - docker
+
+lint build environment test:
+  <<: *parameters_common
+  image: registry.hub.docker.com/koalaman/shellcheck-alpine:stable
+  stage: lint
+  script:
+  - shellcheck -C -f tty -s sh tests/buildenv.sh
+  - shellcheck -C -f tty -s sh build_for_ultimaker.sh
+
+# Build environment
+# =================
+.build environment template: &buildenv_common
+  <<: *parameters_common
+  image: registry.hub.docker.com/library/docker:stable
+  only:
+    variables:
+    - $CI_COMMIT_MESSAGE =~ /dockerfile/i
```

Comment:
It will _only_ run this job if the variable CI_COMMIT_MESSAGE contains (matches, incasesensitive) the regular expression /dockerfile/i

however ideally I wanted to do a /^Dockerfile\:\ /i in only the title, but that breaks merges. What we want is to check if the list of files contains the dockerfile.

https://docs.gitlab.com/ee/ci/yaml/#only-and-except-complex
and
https://gitlab.com/gitlab-org/gitlab-ce/issues/48076